spec/domains/ranks/insanity_scores_spec.rb

```diff
@@ -22,26 +26,55 @@
   end
 
-  describe "tie-breaking" do
+  describe "when scores are tied" do
     let(:input) do
       [
         { score: 50, user_id: pilou.id },
-        { score: 50, user_id: aquaj.id }
+        { score: 50, user_id: aquaj.id },
+        { score: 50, user_id: foo.id }
       ]
     end
 
-    let!(:completions) do
-      [
-        create(:completion, user: aquaj, day: 1, completion_unix_time: Aoc.begin_time + 20.hours),
-        create(:completion, user: pilou, day: 1, completion_unix_time: Aoc.begin_time + 10.hours)
-      ]
+    context "and users have completed different numbers of puzzles" do
+      it "the more completed puzzles the better" do
+        create_list(:completion, 2, user: pilou)
+        create_list(:completion, 3, user: aquaj)
+        create_list(:completion, 1, user: foo)
+
+        expect(described_class.new(input).rank).to eq([
+                                                        input[1],
+                                                        input[0],
+                                                        input[2]
+                                                      ])
+      end
     end
 
-    it "prioritizes shorter completion times" do
-      expect(described_class.new(input).rank).to match([
-                                                         input[0],
-                                                         input[1]
-                                                       ])
+    context "and users have completed the same number of puzzles, but took different times to complete them" do
+      it "prioritizes by smaller total duration of puzzle solving" do
+        create(:completion, user: pilou, completion_unix_time: fake_now + 40.hours)
+        create(:completion, user: aquaj, completion_unix_time: fake_now + 20.hours)
+        create(:completion, user: foo, completion_unix_time: fake_now + 30.hours)
+
+        expect(described_class.new(input).rank).to eq([
+                                                        input[1],
+                                                        input[2],
+                                                        input[0]
+                                                      ])
+      end
+    end
+
+    context "and users have completed the same number of puzzles, took the same time to complete them" do
+      it "prioritizes by user id" do
+        create(:completion, user: pilou, completion_unix_time: fake_now + 2.hours)
+        create(:completion, user: aquaj, completion_unix_time: fake_now + 2.hours)
+        create(:completion, user: foo, completion_unix_time: fake_now + 2.hours)
+
+        expect(described_class.new(input).rank).to eq([
+                                                        input[0],
+                                                        input[1],
+                                                        input[2]
+                                                      ])
+      end
```

Comment:
There seems to be something wrong with the SQL you made to compute the `completions.duration`
When I create two instances of a Completion with the same `completion_unx_time` I can see this attribute is the same for both and yet I have many days of difference between both durations
```bash
[8] pry(#)> create(:completion, completion_unix_time: fake_now + 2.hours)
=> #<Completion:0x00007f28a3eb5280 id: 464, completion_unix_time: 4102470000, duration: 27778 days and 2 hours>
[9] pry(#)> create(:completion, completion_unix_time: fake_now + 2.hours)
=> #<Completion:0x00007f28a3e04750 id: 465, completion_unix_time: 4102470000, duration: 27790 days and 2 hours>
```

Reply:
C'est ca qui casse les deux tests si je change et que je tri par `completion_uniw_time` a la place les tests passent
```ruby
        -completions.sum { |completion| completion.completion_unix_time || 0 },
```

Reply:
Oh wow, bien vu pour le `duration` qui change :O Je vais investiguer, merci d'avoir jeté un oeil :pray:

Reply:
Normalement c'est résolu :tada: 

La factory qui crée les `completions` prend une valeur pour le `day` et le `challenge` qui est aléatoire par défaut. Du coup, sans spécifier de jour, la completion créée peut correspondre au jour  `10` ou au jour `15`, ce qui fait mécaniquement varier la `duration` à `completion_unix_time` constant. Et des tests qui un coup passent, un coup ne passent pas.

J'ai donc fixé le jour pour les complétions créées là où c'est nécessaire, tous les tests passent :+1: